doc/design/doi-generation-2.md

```diff
@@ -0,0 +1,66 @@
+- `Registered`. Like `Findable` but not indexed for search, so we do not use them.
+- `Findable`. Is the type we use for published dandisets.
+  Requires to be valid (pass validation to fit the datacite schema) to be created.
+
+We propose to:
+
+- Instead of a fake DOI, upon creation of dandiset, mint and use a legit `Draft` DOI `10.48324/dandi.{dandiset.id}` with
+ - metadata entered during creation request (title, description, license)
+ - DLP URL `https://dandiarchive.org/dandiset/{dandiset.id}`
+ - For embargoed dandiset, **do not** specify any metadata besides the DLP URL.
+- Upon changes to dandiset metadata record, for public (non-embargoed dandisets), try to update datacite metadata record while keeping the same target URL.
+  - For Draft DOI (dandiset was not published yet), there is no validation.
+    - **Question to clear up**: what happens to Draft DOI if metadata record is invalid? Does it fail to update altogether? does it update only the fields it knows about?
+  - For Findable DOI (dandiset was published), metadata record must pass validation, so we might fail to update.
+    But that should be ok.
+- Upon unembargoing dandiset: update (Draft) DOI metadata record with current metadata.
+- Upon publication of the dandiset:
+  - (already done currently) mint a proper `Findable` version DOI `10.48324/dandi.{dandiset.id}/{version}`
+  - update Dandiset-wide DOI `10.48324/dandi.{dandiset.id}` with metadata provided for version DOI, while keeping URL pointing to DLP instead of the released version.
+    - if Dandiset-wide DOI was in Draft state, it would be updated to Findable state (should work since we know metadata record passed validation).
+
+## Concerns to keep in mind/address
+
+- Draft dandiset might not have sufficient metadata to mint a proper DOI, or metadata might not be "proper" (fail validation) thus causing issues with minting a DOI
+  - Solution: start with Draft (not findable) DOI, and then upon publication mint a "findable" DOI
+  - **Follow up concern**: after dandiset and DOI publish, metadata of the Draft version of the dandiset could still be changed.
+    This potentially making changed record again "invalid".then when people change metadata
+- test site of datacite had different result of validation that the primary one
+
+- `Findable` DOI cannot be deleted, but in principle we allow for deletion of dandisets.
+  - We might want a dedicated 404 page for deleted dandisets, or at least a message that the dandiset was deleted, and ideally describe the reason why it was deleted ("Upon request of maintainer", "Due to violation of terms of service", etc.)
+  - Then we adjust DOI record to point to that page.
+
+- Should we do anything at dandischema level?
+
+- Should we do anything at DLP level?
+
+- Should we somehow reflect interactions with DataCite in Audit log?
```

Comment:
```suggestion
- Should we somehow reflect interactions with DataCite in Audit log?

# Targets

- [ ] develop a script, which tests on test fabric of datacite changes as introduced to all dandisets in the archive by 
     - for each dandiset
       - generate a record for overall *dandiset DOI* corresponding to metadata of the first release if any exists, otherwise corresponding to metadata of the draft version
       - for each release: mint a new *version DOI* for that release + possibly update *dandiset DOI* to correspond to potential changes in metadata
       - update *dandiset DOI*  to metadata of draft version
- implementation in dandi-archive
```